Core messaging implementation.

LMiljak opened the pull request:
-Added an extra module ContextMessages that both other two modules have as dependency (for it to work you have to run "mvn install" on the ContextProject)
-The module contains an abstract MessageListener class that should be extended in place of the JME3 MessageListener interface, because the MessageListener class only handles one specific message instead of multiple (which is bad practice).
-Accelerator (gyroscope) functionality has been moved to a separate class.

marcelkuipers committed:
moved gyroscope functionality to different class

Commit:
Added ContextMessages Module.

Both the ContextDesktop and ContextApp have this module as a dependency,
meaning that both of them can use every class in this module.

Commit:
Made the ContextMessage module run with Java 1.8

Commit:
Added a MessageListener class.

See the JavaDoc of the class for info.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature_AndroidMessaging

Commit:
Added tests for the MessageListener class.

Commit:
ServerWrapper now supports messages.

For every message that the server should handle: The message type
(message.class) should be put in the MESSAGE_TYPES field.

For every message listener that the server should handle: The message
listener should be put in the messageListeners field.

Commit:
Some Pom layout changes.

Commit:
Removed the messageListeners field in the Wrappers

Instead of that,
INSTANCE.getServer()/getClient().addMessageListener(messageListener)
should be called by the messageListener itself.

Commit:
Removed unnecessary warning supressor.

Commit:
Ignoring the /ContextMessages/target folder.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature_AndroidMessaging

Commit:
ContextMessages Module runs on 1.7

This is so that the ContextApp module (that has to run on 1.7) can make
use of this module as well.

The mayor change is that the MessageListener interface is now an
abstract class, because Java 1.7 does not support default methods.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature_AndroidMessaging

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature_AndroidMessaging

NilsHullegien commented on ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java:
```diff
@@ -0,0 +1,53 @@
+ *
+ * @author Marcel
+ */
+public class AccelerometerSensor extends Activity implements SensorEventListener {
```
Missing class javadoc

NilsHullegien commented on ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java:
```diff
@@ -74,34 +74,39 @@ public void onCreate(Bundle savedInstanceState) {
+//                    return null;
+//                }
+//             });
+//        }
```
Please remove this.

NilsHullegien commented on ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java:
```diff
@@ -74,34 +74,39 @@ public void onCreate(Bundle savedInstanceState) {
+//        @Override
+//        public void onAccuracyChanged(Sensor sensor, int i) {
+//        }
+//        
```
Please remove this.

NilsHullegien commented:
It seems like there are still some PMD, and CheckStyle errors. 
Please fix these as well.

Commit:
Removed useless comments.

Commit:
Added CheckStyle files in Messages module

These should have been added when the module was created, but didn't
happen.

Commit:
Added pmd files in Messages module

These should have been added when the module was created, but it didn't
happen.

Commit:
Fixed CheckStyle violations in TestMessageListener

Commit:
CheckStyle, pmd and FindBugs are reported.

Commit:
Fixed FindBugs warning in TestMessageListener.

Commit:
Fixed CheckStyle warnings in ContextApp.

MrCrime commented:
@NilsHullegien 
Appearently the ContextMessages module didn't even have the correct CheckStyle and pmd files in it. I've added those and fixed the errors they gave.
Also removed those useless comments pointed out by you.

NilsHullegien commented:
Looks good now, thanks for adding the CheckStyle and pmd in the ContextMessages module as well.

Commit:
Fixed Messages module breaking App module.

The ContextMessages module contained plugins in the pom that broke the
ContextApp.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature_AndroidMessaging

# Conflicts:

#	ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java

MrCrime commented:
Fixed merge conflict with master.

indisputable commented on ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java:
```diff
@@ -0,0 +1,58 @@
+    public AccelerometerSensor(HelloActivity act) {
+        this.act = act;
+    }
+    
```
Please add javadoc

Commit:
Removed plugin in the pom of App module

This plugin is meant for the Desktop module, not the App module. How
this got in the App pom in the first place is a mystery.

dVoorhout committed:
javadaoc

indisputable commented:
everything else looks fine
